src/config.c

```diff
@@ -882,23 +912,22 @@ void configSetCommand(client *c) {
                 addModuleConfigApply(module_configs_apply, config->privdata);
             } else if (config->interface.apply) {
                 /* Check if this apply function is already stored */
-                config_name = sdsnew(config->name);
-                if (!(dictFind(apply_fns, config_name))) {
-                    dictAdd(apply_fns, sdsnew(config->name), &config->interface);
+                if (!(dictFind(apply_fns, &config->interface))) {
+                    dictAdd(apply_fns, &config->interface, sdsnew(config->name));
```

Comment:
The key needs to be the apply function itself, not the config/interface.
```suggestion
                if (!(dictFind(apply_fns, &config->interface.apply))) {
                    dictAdd(apply_fns, config->interface.apply, sdsnew(config->name));
```

Reply:
I intended on doing this initially but doing this gives the following warning
`warning: ISO C forbids initialization between function pointer and ‘void *’`